src/main/java/de/tum/in/www1/artemis/service/connectors/pyris/PyrisWebhookService.java

```diff
@@ -0,0 +1,172 @@
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.Base64;
+import java.util.List;
+import java.util.NoSuchElementException;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.context.annotation.Profile;
+import org.springframework.stereotype.Service;
+
+import de.tum.in.www1.artemis.aop.logging.LoggingAspect;
+import de.tum.in.www1.artemis.domain.Course;
+import de.tum.in.www1.artemis.domain.enumeration.AttachmentType;
+import de.tum.in.www1.artemis.domain.iris.settings.IrisCourseSettings;
+import de.tum.in.www1.artemis.domain.lecture.AttachmentUnit;
+import de.tum.in.www1.artemis.repository.iris.IrisSettingsRepository;
+import de.tum.in.www1.artemis.service.FilePathService;
+import de.tum.in.www1.artemis.service.connectors.pyris.dto.PyrisPipelineExecutionSettingsDTO;
+import de.tum.in.www1.artemis.service.connectors.pyris.dto.lectureingestionwebhook.PyrisLectureUnitWebhookDTO;
+import de.tum.in.www1.artemis.service.connectors.pyris.dto.lectureingestionwebhook.PyrisWebhookLectureIngestionExecutionDTO;
+import de.tum.in.www1.artemis.service.iris.exception.IrisInternalPyrisErrorException;
+import de.tum.in.www1.artemis.service.iris.settings.IrisSettingsService;
+
+@Service
+@Profile("iris")
+public class PyrisWebhookService {
+
+    private static final Logger log = LoggerFactory.getLogger(LoggingAspect.class);
+
+    private final PyrisConnectorService pyrisConnectorService;
+
+    private final PyrisJobService pyrisJobService;
+
+    private final IrisSettingsService irisSettingsService;
+
+    @Value("${server.url}")
+    private String artemisBaseUrl;
+
+    public PyrisWebhookService(PyrisConnectorService pyrisConnectorService, PyrisJobService pyrisJobService, IrisSettingsService irisSettingsService) {
+        this.pyrisConnectorService = pyrisConnectorService;
+        this.pyrisJobService = pyrisJobService;
+        this.irisSettingsService = irisSettingsService;
+    }
+
+    private boolean lectureIngestionEnabled(Course course) {
+        return irisSettingsService.getRawIrisSettingsFor(course).getIrisLectureIngestionSettings() != null
+                && irisSettingsService.getRawIrisSettingsFor(course).getIrisLectureIngestionSettings().isEnabled();
+    }
+
+    private String attachmentToBase64(AttachmentUnit attachmentUnit) {
+        Path path = FilePathService.actualPathForPublicPathOrThrow(URI.create(attachmentUnit.getAttachment().getLink()));
+        try {
+            byte[] fileBytes = Files.readAllBytes(path);
+            return Base64.getEncoder().encodeToString(fileBytes);
+        }
+        catch (IOException e) {
+            throw new IrisInternalPyrisErrorException(e.getMessage());
+        }
+    }
+
+    private PyrisLectureUnitWebhookDTO processAttachmentForUpdate(AttachmentUnit attachmentUnit) {
+        int lectureUnitId = attachmentUnit.hashCode();
```

Comment:
Why don't you use the actual id?

Reply:
I thought it would be better to use the hashcode, in terms of safety. But I am going to change it, as I was reluctant between both approaches